### PR TITLE
Move to stdlib and expand taint-tracking for `io` and `io/ioutil` packages

### PR DESCRIPTION
**Pull request by gagliardetto**
`codebox` commands:

```bash
codebox --out-dir=./generated/latest --pkg=/usr/local/go/src/io --http
codebox --out-dir=./generated/latest --pkg=/usr/local/go/src/io/ioutil --http
```
---

Part of #167

### CODE REVIEW

**Comment by gagliardetto** [comment]
I removed the byte/rune read/write taint tracking; is that ok?

**Comment by smowton** [comment]
Sounds good to me

**Comment by smowton** [comment]
I note the frontend errors consistency check is failing again here -- try a `go build` in the test directory with Go 1.14 and I suspect you'll find the trouble.

File: ql/src/semmle/go/frameworks/stdlib/Io.qll [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+      // signature: func (*SectionReader).Read(p []byte) (n int, err error)
+      this.hasQualifiedName("io", "SectionReader", "Read") and
+      (inp.isReceiver() and outp.isParameter(0))
+      or
+      // signature: func (*SectionReader).ReadAt(p []byte, off int64) (n int, err error)
+      this.hasQualifiedName("io", "SectionReader", "ReadAt") and
+      (inp.isReceiver() and outp.isParameter(0))
+      or
+      // signature: func (Reader).Read(p []byte) (n int, err error)
+      this.implements("io", "Reader", "Read") and
```
Comment (smowton) [inline ql/src/semmle/go/frameworks/stdlib/Io.qll:97]: Are the PipeReader.Read etc rules redundant because of this? Similarly ReaderAt and so on?

Comment (gagliardetto) [reply ql/src/semmle/go/frameworks/stdlib/Io.qll:77]: Removed from io package. Will do in another PR for all the other packages.

**Review by smowton — APPROVED** [review]

**Pull request merged**